### DISCUSSION TITLE
sendRC6(raw,numBits) works, but sendRC6(addr,cmd,numRepeats) doesn't work

### DISCUSSION THREAD

**Discussion by DaveLowther56** [Arduino-IRremote/Arduino-IRremote#1051] [Help] [answered]
Using v3.9.0 of the library with Arduino Uno.
I am sending channel number keys to a set top box.
'number' is in the range 0..9.
I also have receiver running in the same Arduino. 

If I send with this line (when number=0)
  IrSender.sendRC6((uint32_t)0xC05C00UL + number, 24);

The set top box responds correctly and after IrReceiver.decode() debug prints

>>>>>>>>>>>
Fields of IrReceiver.decodedIRData ...
  protocol       = 15
  address        = 0x5C
  command        = 0x0
  extra          = 0x0
  numberOfBits   = 24
  flags          = 0x80
  decodedRawData = 0xC05C00
Send with: IrSender.sendRC6(0x5C, 0x0, <numberOfRepeats>);
<<<<<<<<<<<

If I send with this line (when number=0)
  IrSender.sendRC6(0x5C, number, 0);

The set top box does not respond and after IrReceiver.decode() debug prints

>>>>>>>>>>>
Fields of IrReceiver.decodedIRData ...
  protocol       = 15
  address        = 0x5C
  command        = 0x0
  extra          = 0x0
  numberOfBits   = 20     <----- not 24
  flags          = 0x80
  decodedRawData = 0x5C00 <----- missing leading 'C0'
Send with: IrSender.sendRC6(0x5C, 0x0, <numberOfRepeats>);
<<<<<<<<<<<

Can someone tell me what I am doing wrong?

Edit 
![image](https://user-images.githubusercontent.com/116661947/204277386-9b66b064-88d1-4a16-b244-5bda5cca5820.png)
I assume markup removed the numberOfRepeats section

**Comment by ArminJo** [comment, marked as answer]
I improved the RC6 decoder and the printIRSendUsage() function to be more specific.
What you have is **a no standard RC6 protocol**, a protocol derived from RC6 using more bits. Therfore you can use the sendRC6(uint32_t...) function - now renamed to sendRC6Raw() - to send this non standard RC6 like protocol.

**Reply by DaveLowther56** [reply]
Thank you very much for your help. It's much appreciated.
